src/content-helper/dashboard-widget/provider.ts

```diff
@@ -0,0 +1,104 @@
+/**
+ * External dependencies
+ */
+import { addQueryArgs } from '@wordpress/url';
+import apiFetch from '@wordpress/api-fetch';
+
+/**
+ * Internal dependencies
+ */
+import {
+	ContentHelperError,
+	ContentHelperErrorCode,
+} from '../../blocks/content-helper/content-helper-error';
+import { TopPostData } from './top-posts/model';
+import {
+	convertDateToString,
+	removeDaysFromDate,
+} from '../../blocks/shared/utils/date';
+
+/**
+ * The form of the response returned by the /stats/posts WordPress REST API
+ * endpoint.
+ */
+interface TopPostsApiResponse {
+	error?: Error;
+	data?: TopPostData[];
+}
+
+/**
+ * The form of the result returned by the getTopPosts() function.
+ */
+export interface GetTopPostsResult {
+	posts: TopPostData[];
+}
+
+export const TOP_POSTS_DEFAULT_LIMIT = 3;
+export const TOP_POSTS_DEFAULT_TIME_RANGE = 7; // In days.
```

Comment:
For related posts it is `3 days` here we are going with `7 days` should we make it consistent?

If we are making this consistent then should we define this in some common location and use from that or should we just enforce this by default on our API endpoint?

Reply:
I'd flag this in our internal communications, and we can get back to it if needed in another PR. It is a small change to make if we need to. This is as per the requirements [here](https://github.com/Parsely/wp-parsely/issues/1221#issuecomment-1353319380)